### ISSUE TITLE
QueryFailedError: Corrupted alias

### ISSUE THREAD

**Issue by whytekieran** [oguimbal/pg-mem#93]
Ive recently been using the pg-mem package to perform some tests using jest and typeorm.

I recently performed a typeorm migration and now I'm getting a "corrupted alias" exception. It seems to think one of the columns in the table is the old column name before the migration. Any ideas how i can resolve this? Attaching test code, was working fine before running the migration. Entity name is Catalog. In last migration i changed column code to catalogCode. Since then test wont work. Also catalogCode is an autogenerated uuid hence my implementation of registerFunction()

```import { join } from 'path';
import { DataType, IMemoryDb, newDb } from 'pg-mem';
import { Connection, Repository } from 'typeorm';
import { Catalog } from './models/catalog.entity.model';

describe('Catalog TypeORM PostGreSQL Integration Test', () => {
  // Ideally, Integration Tests should be runned with a Docker instance of PostGreSQL
  let db: IMemoryDb;
  let catalog: Repository<Catalog>;

  beforeAll(async () => {
    // Setup DB Config. Recommended when using Typeorm .synchronize(), which creates foreign keys but not indices !
    db = newDb({
      autoCreateForeignKeyIndices: false,
    });

    // Function uuid_generate_v4 must be registered with pg-mem, set to return a hardcoded uuid for test purposes
    db.public.registerFunction({
      name: 'uuid_generate_v4',
      args: [],
      returns: DataType.uuid,
      implementation: () => '0421a7c8-862f-4152-84fe-3a8b7d3bc805',
    });

    const migrationsPath = join(__dirname, '../../../src/**/migrations/**/*{.ts,.js}');

    // Create a Typeorm connection
    const connection: Connection = await db.adapters.createTypeormConnection({
      type: 'postgres',
      entities: [Catalog],
      migrations: [migrationsPath],
      migrationsRun: true,
    });

    // Create tables
    catalog = connection.getRepository(Catalog);
  });

  it('should save and find new catalog', async () => {
    await catalog.save({
      title: 'title 1',
      isPublic: false,
      isDefault: false,
    });
    await catalog.save({
      title: 'title 2',
      isPublic: false,
      isDefault: false,
    });
    const foundCatalogs = await catalog.find();
    expect(foundCatalogs).toHaveLength(2);
  });
});
```

Stacktrace:
at new QueryFailedError (src/error/QueryFailedError.ts:9:9)
      at PostgresQueryRunner.<anonymous> (src/driver/postgres/PostgresQueryRunner.ts:228:19)
      at step (node_modules/typeorm/node_modules/tslib/tslib.js:141:27)
      at Object.throw (node_modules/typeorm/node_modules/tslib/tslib.js:122:57)
      at rejected (node_modules/typeorm/node_modules/tslib/tslib.js:113:69)

**Comment by oguimbal** [comment]
Hi !

- What is the exact version of pg-mem that is installed (as seen in `cat ./node_modules/pg-mem/package.json | grep version` ?
- Same question for typeorm: Which exact version ?
- The error message should contain the failing request... could you provide it ?

Thanks

**Comment by whytekieran** [comment]
Hi,

Dependencies found in package.json:
```
"pg-mem": "^1.8.6", (same when running cat command 1.8.6)
"typeorm": "0.2.30",
"@nestjs/typeorm": "7.1.5",
```
This is entire output i get in my stack trace, is that what you need?

```QueryFailedError: Corrupted alias

    �� This is a nasty error, which was unexpected by pg-mem. Also known "a bug" �� Please file an issue !

    *️⃣ Failed SQL statement: INSERT INTO "catalog"("title", "catalogCode", "isPublic", "isDefault") VALUES ('title 1', '9afe1960-4725-4608-95fe-f3c257654ff
2', false, false) RETURNING "id", "catalogCode", "isPublic", "isDefault";

    �� You can file an issue at https://github.com/oguimbal/pg-mem along with a way to reproduce this error (if you can), and  the stacktrace:

      at new QueryFailedError (src/error/QueryFailedError.ts:9:9)
      at PostgresQueryRunner.<anonymous> (src/driver/postgres/PostgresQueryRunner.ts:228:19)
      at step (node_modules/typeorm/node_modules/tslib/tslib.js:141:27)
      at Object.throw (node_modules/typeorm/node_modules/tslib/tslib.js:122:57)
      at rejected (node_modules/typeorm/node_modules/tslib/tslib.js:113:69)```

**Comment by oguimbal** [comment]
First, there are known issues with Typeorm > 0.2.29 ... could you try downgrading to 0.2.29 ?

That said, the usual error with typeorm 0.2.30 is not this one (see #53), yours is quite hard to explain:

- The given stacktrace is the stack as rethrown by Typeorm (if you can attach a debugger with "break all exception": what is the actual stack trace of the error when it is originally thrown ?)
- The statement that appears to be failing does not contain any alias... My best guess is that is triggers another inner request, or something like that (and that the actual failing request is hidden by the outer one)

So in short, I cant reproduce it with the information i have, sorry :(

![image](https://user-images.githubusercontent.com/8973947/114670406-3daad280-9d03-11eb-853d-70afdaf1b5a5.png)

Could you try gathering more intel ?
Like the actual stacktrace mentioned above, or setup a reproduction repo I can access ?

**Comment by whytekieran** [comment]
Hi,

I decided to try a clean slate. I removed all migration files and created just one which creates entire table (no modification files)

**Comment by oguimbal** [comment]
No problem.. I'm closing this, then ! Reopen if needed
